01_Data/src/layers/sequelize/mapper/2.0.1/AuthenticationMapper.ts

```diff
@@ -0,0 +1,125 @@
+import {
+    OCPP2_0_1
+} from "@citrineos/base";
+import {AdditionalInfo, Authorization, IdToken} from "../../model/Authorization";
+
+export class AuthenticationMapper {
+
+    static toAuthorizationData(authorization: Authorization): OCPP2_0_1.AuthorizationData {
+        return {
+            customData: authorization.customData,
+            idToken: AuthenticationMapper.toIdToken(authorization.idToken),
+            idTokenInfo: AuthenticationMapper.toIdTokenInfo(authorization),
+        }
+    }
+
+    static toIdToken(idToken: IdToken) {
+        return {
+            customData: idToken.customData,
+            additionalInfo: idToken.additionalInfo && idToken.additionalInfo.length > 0
+                ? (idToken.additionalInfo.map(this.toAdditionalInfo) as [any, ...any[]])
+                : null,
+            idToken: idToken.idToken,
+            type: AuthenticationMapper.toIdTokenEnumType(idToken.type),
+        }
+    }
+
+    static toAdditionalInfo(additionalInfo: AdditionalInfo) {
+        return {
+            customData: additionalInfo.customData,
+            additionalIdToken: additionalInfo.additionalIdToken,
+            type: additionalInfo.type
+        }
+    }
+
+    static toIdTokenInfo(authorization: Authorization) {
```

Comment:
It would be perfect to add the return type for these mapping methods. 
```suggestion
    static toIdTokenInfo(authorization: Authorization): OCPP2_0_1.IdTokenInfoType {
```

Reply:
The thing is with 1.6 mapper, we wouldn't be able to do such thing since we don't have access to the nested objects so I left out doing so here for sake of consistency.